lib/iris/coords.py

```diff
@@ -2674,19 +2619,20 @@ def __init__(self, method, coords=None, intervals=None, comments=None):
                 "'method' must be a string - got a '%s'" % type(method)
             )
 
-        default_name = CFVariableMixin._DEFAULT_NAME
+        default_name = BaseMetadata.DEFAULT_NAME
```

Comment:
@lbdreyer It should be public, as the `iris.coords.CellMethod` requires access to it. 

If it was private, then the side-effect is that as a developer I might want to change the value of `BaseMetadata.DEFAULT_NAME` thinking that it's totally safe to do so, and yet it would directly impact the behaviour of `iris.coords.CellMethod`.

If `DEFAULT_NAME` was only used internally within `BaseMetadata` then I'd suggest that it's safe to make it private - but that's not the case. Hence the change.